src/components/auth/PrivateRoute.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import {useSelector} from 'react-redux';
+import {Route, Redirect, RouteProps} from 'react-router-dom';
+import {RootState} from '../../store';
+
+import AppPaths from '../../const/paths';
+
+interface Props extends RouteProps {
+    component: any;
+}
+
+export const PrivateRoute = ({component: Component, ...rest}:Props) => {
+  const {authenticated} = useSelector((state: RootState) => state.auth);
```

Comment:
not sure why authenticated is being passed in brackets like this you shouldn't have to do that I believe. Also if you are doing a check on authentication be more specific and point towards the `state.auth.authenticated` boolean.

Reply:
Yeah, I think that's exactly what it's doing. The data must be taken from the store since that's where the state is stored. It's grabbing auth state straight from the store. And as far as destructuring, that's done so that we can do assignment all in one take.

Reply:
Also, if you try without destructuring, the variable will still cause the user to be authenticated even when they should not be. So destructuring is necessary.